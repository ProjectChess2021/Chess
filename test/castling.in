setup
+ k e8
+ r h8
+ K e5
+ R a8
+ R a1
+ p e7
+ P h3
a8 is ever moved
= black
done
game human human
move e8 g8
move e1 g1
move e1 c1