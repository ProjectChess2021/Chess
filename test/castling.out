8  _ _ _ _      MoveHistory
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ _ _ 

  abcdefgh
8  _ _k_ _      MoveHistory
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ _ _ 

  abcdefgh
8  _ _k_ r      MoveHistory
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ _ _ 

  abcdefgh
8  _ _k_ r      MoveHistory
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _ 

  abcdefgh
8  _ _k_ r      MoveHistory
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ _ K _R

  abcdefgh
8  _ _k_ r      MoveHistory
7 _ _ _ _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 R _ K _R

  abcdefgh
8  _ _k_ r      MoveHistory
7 _ _ p _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 R _ K _R

  abcdefgh
8  _ _k_ r      MoveHistory
7 _ _ p _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _P
2  _ _ _ _
1 R _ K _R

  abcdefgh
a8 is set to be ever moved.
Setup successful and it's black's turn to go next.
8  _ _k_ r      MoveHistory
7 _ _ p _ 
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 R _ K _R

  abcdefgh
8  _ _ rk_      MoveHistory
7 _ _ p _       Black:  king e8 -> g8 (castling)
6  _ _ _ _
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 R _ K _R

  abcdefgh
Rook is ever moved thus cannot do that.
8  _ _ rk_      MoveHistory
7 _ _ p _       Black:  king e8 -> g8 (castling)
6  _ _ _ _      White:  king e1 -> c1 (castling)
5 _ _ _ _ 
4  _ _ _ _
3 _ _ _ _ 
2  _ _ _ _
1 _ KR_ _R

  abcdefgh
